lgc/state/PipelineState.cpp

```diff
@@ -683,13 +683,41 @@ const ResourceNode *PipelineState::findPushConstantResourceNode() const {
 }
 
 // =====================================================================================================================
-// Find the resource node for the given {set,binding}.
-// For nodeType == Unknown, the function finds any node of the given set,binding.
-// For nodeType == Resource, it matches Resource or CombinedTexture.
-// For nodeType == Sampler, it matches Sampler or CombinedTexture.
-// For nodeType == Buffer, it matches Buffer, BufferCompact or InlineBuffer.
-// For other nodeType, only a node of the specified type is returned.
-// For nodeType == DescriptorTableVaPtr, the node is returned whose first child matches descSet.
+// Returns true when type nodeType is compatible with candidateType.
+// A node type is compatible with a candidate type iff (nodeType) <= (candidateType) in the ResourceNodeType lattice:
+//
+//                                                        DescriptorCombinedTexture
+//                                                                   +
+// DescriptorBufferCompact   InlineBuffer                            |
+//                   +         +                 +-------------------+--------------------+
+//                   |         |                 |                   |                    |
+//                   v         v                 v                   v                    v
+//                 DescriptorBuffer     DescriptorResource  DescriptorTexelBuffer  DescriptorSampler
+//                         +                     +                   +                    +
+//                         |                     |                   |                    |
+//                         |                     v                   |                    |
+//                         +----------------> Unknown <--------------+--------------------+
+//
+static bool IsNodeTypeCompatible(ResourceNodeType nodeType, ResourceNodeType candidateType) {
```

Comment:
Please follow coding style to add parameter comments even if their meanings are obvious.

//
// @param nodeType : ...
// @param candidateType : ...

Reply:
Done. IMO we should consider relaxing this for local helper functions, as this disincentivizes factoring out of large functions, but that's a topic for a separate discussion.

Reply:
Yes, this is another topic to revise the coding style and introduce some feasibility. We can discuss the practical rules